metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery_schema.py

```diff
@@ -157,21 +157,19 @@ def get_projects(self) -> List[BigqueryProject]:
                 # Also project.list returns max 50 projects per page.
                 # Assuming list_projects method internally not adding any limit in requests call,
                 # externally we are adding limit in requests call per second.
+                rate_limiter = RateLimiter(max_calls=2, period=1)
                 while True:
-                    projects_iterator = self.bq_client.list_projects(
-                        max_results=50, page_token=page_token
-                    )
-                    projects.extend(
-                        [
-                            BigqueryProject(id=p.project_id, name=p.friendly_name)
-                            for p in projects_iterator
-                        ]
-                    )
+                    with rate_limiter:
+                        projects_iterator = self.bq_client.list_projects(
+                            max_results=30, page_token=page_token
```

Comment:
what is the max number of results in a single page? can you add a link to the docs here?

Reply:
Its 50 projects per page
https://cloud.google.com/bigquery/docs/reference/rest/v2/projects/list#query-parameters